app/models/mixins/compliance_mixin.rb

```diff
@@ -4,7 +4,7 @@ module ComplianceMixin
   included do
     has_many :compliances, :as => :resource, :dependent => :destroy
     has_one  :last_compliance,
-             -> { order('"compliances"."timestamp" DESC') },
+             -> { order(Arel.sql('"compliances"."timestamp" DESC')) },
```

Comment:
```suggestion
             -> { order(:timestamp => 'DESC') },
```

```
Compliance.order(:timestamp => 'DESC').take(5)

  Compliance Load (47.9ms)  SELECT  "compliances".* FROM "compliances" ORDER BY "compliances"."timestamp" DESC LIMIT $1  [["LIMIT", 5]]
```

Reply:
Does it matter that this is on the has_one `:last_compliance` and uses the compliances?  Can you run your query on a class that has many compliances?

Reply:
before:

```
ExtManagementSystem.first.last_compliance

  ExtManagementSystem Load (43.9ms)  SELECT  "ext_management_systems".* FROM "ext_management_systems" ORDER BY "ext_management_systems"."id" ASC LIMIT $1  [["LIMIT", 1]]
  ExtManagementSystem Inst Including Associations (11604.9ms - 1rows)
  Compliance Load (27.4ms)  SELECT  "compliances".* FROM "compliances" WHERE "compliances"."resource_id" = $1 AND "compliances"."resource_type" = $2 ORDER BY "compliances"."timestamp" DESC LIMIT $3  [["resource_id", 1], ["resource_type", "ExtManagementSystem"], ["LIMIT", 1]]
```

after:

```
ExtManagementSystem.first.last_compliance
  ExtManagementSystem Load (0.6ms)  SELECT  "ext_management_systems".* FROM "ext_management_systems" ORDER BY "ext_management_systems"."id" ASC LIMIT $1  [["LIMIT", 1]]
  ExtManagementSystem Inst Including Associations (7300.6ms - 1rows)
  Compliance Load (0.6ms)  SELECT  "compliances".* FROM "compliances" WHERE "compliances"."resource_id" = $1 AND "compliances"."resource_type" = $2 ORDER BY "compliances"."timestamp" DESC LIMIT $3  [["resource_id", 1], ["resource_type", "ExtManagementSystem"], ["LIMIT", 1]]
  Compliance Inst Including Associations (0.0ms - 0rows)
```